lib/src/layout/sidebar/sidebar_items.dart

```diff
@@ -190,16 +187,15 @@ class _SidebarItem extends StatelessWidget {
     Key? key,
     required this.item,
     required this.onClick,
-    required this.selected,
+    required this.currenIdentifier,
   }) : super(key: key);
 
   /// The widget to lay out first.
   ///
   /// Typically an [Icon]
   final SidebarItem item;
 
-  /// Whether the item is selected or not
-  final bool selected;
+  final String? currenIdentifier;
```

Comment:
Why the change from a boolean to the current identifier here? Seems like the selected state could still be computed where this class is instantiated.

Reply:
Setting this to an identifier allows us to infer the selection state at any level of the sidebar items tree.